### PR TITLE
Update Developer Guide by adding Edit and Delete commands

### LINKED ISSUE #209: DG: Wrong Arrow
image

### LINKED ISSUE #245: DG: Update AB3 diagrams

### PR DESCRIPTION
**Pull request by Respirayson**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -122,18 +122,11 @@ How the parsing works:
 
 The `Model` component,
 
-* stores the address book data i.e., all `Person` objects (which are contained in a `UniquePersonList` object).
-* stores the currently 'selected' `Person` objects (e.g., results of a search query) as a separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
-* stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
+* stores the address book data i.e., all `Person`, `Task`, `Session`, `GradedTest` and `Consultation` objects (which are contained in a `UniquePersonList`, `TaskList`, `SessionList`, `GradedTestList` and `ConsultationList` objects respectively).
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: F.A.K.E.J.A.R.V.I.S.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -329,6 +323,14 @@ Step 7:
 The `XYZCommand` creates a successful `CommandResult` and returns it to the UI.
 
 
+#### Delete Student Feature
+This section explains the implementation of the Delete Student feature via the `delete` command.
+The `DeleteCommand` causes the specified `Person` to be deleted from the application. This includes removing the student from any `consultatios` or `sessions` that the student might be in.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: typo for `consultations`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -420,12 +420,15 @@ Below is the sequence diagram outlining the execution of `UpdateTasksProgressCom
...
 Step 1:
-The `LogicManager` invokes `UpdateTaskProgressCommand::execute`, which in turn calls `UpdateTaskProgressCommand::createTask` to create a new immutable Task object with the updated progress.
+The `LogicManager` invokes `UpdateTaskProgressCommand::execute`, which in turn calls `Model::getFilteredTaskList` and `List<Task>::get` to get the relevant task to be edited.
 
 Step 2:
-The `UpdateTaskProgressCommand` will invoke `setTask` in `Model`, which in turn calls `setTask` in `TaskListBook`. This finally calls `editTask` in `TaskList` to update the existing `Task` with the new `Task`.
+The `UpdateTaskProgressCommand::createTask` is invoked to create a new immutable Task object with the updated progress.
 
 Step 3:
+The `UpdateTaskProgressCommand` will call `setTask` in `Model` to replace the existing `Task` with the new `Task`.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: replace the existing `Task` object with a new `Task` object

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -420,12 +420,15 @@ Below is the sequence diagram outlining the execution of `UpdateTasksProgressCom
...
+
+Step 4:
 The `UpdateTaskProgressCommand` then invokes `updateFilteredTaskList` in `Model` to display all the tasks.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: Might want to standardise this with those above:

 The `UpdateTaskProgressCommand` will call its own `updateFilteredTaskList` method to update the model's filter and display all the tasks to the user.

File: docs/diagrams/AddTaskSequenceDiagram.puml [hunk 5]
```diff
@@ -7,29 +7,12 @@ participant "x:AddTaskCommand" as AddTaskCommand LOGIC_COLOR
...
-activate TaskList
-
-TaskList ---> TaskListBook
-deactivate TaskList
-
-TaskListBook ---> Model
-deactivate TaskListBook
-
 Model -> Model : updateFilteredTaskList()
 activate Model
```
Comment (raydenlim) [inline docs/diagrams/AddTaskSequenceDiagram.puml:18]: Might want to check this condition:
```java
if (model.hasTask(task)) {
            throw new CommandException(MESSAGE_DUPLICATE_TASK);
}
```

```puml
alt Task does not exist
  AddTaskCommand -> Model: addTask(Task)
  Model --> AddTaskCommand: success message
  AddTaskCommand -> User: success message
else Task already exists
  AddTaskCommand --> User: error message
```

**Review by raydenlim — APPROVED** [review]
Other than a few minor details, LGTM!

File: docs/diagrams/AddTaskSequenceDiagram.puml (hunk 5 above)
Comment (Respirayson) [reply docs/diagrams/AddTaskSequenceDiagram.puml:18]: actually what do the rest think? I drew this sequence diagram with the idea to outline the ideal case at a high level. So, I didnt account for the error checking part in the sequence diagram. Should we standardise and add the error checking part?

**Pull request merged**